### PR TITLE
Clarify slightly description of data_range parameter.

### PR DESCRIPTION
**Pull request by mkcor**
## Description

This tiny PR was inspired by #7342; I read the [docstring](https://scikit-image.org/docs/dev/api/skimage.metrics.html#skimage.metrics.structural_similarity) for `data_range` and thought the term 'distance' wasn't actually clear; maybe I should take this even further and write, outside of any parenthesis, that `data_range` is expected to be a scalar value?

## Checklist

- A descriptive but concise pull request title
- [Docstrings for all functions](https://github.com/numpy/numpy/blob/master/doc/example.py)
- [Unit tests](https://scikit-image.org/docs/dev/development/contribute.html#testing)
- A gallery example in `./doc/examples` for new features
- [Contribution guide](https://scikit-image.org/docs/dev/development/contribute.html) is followed

## Release note

We use [changelist](https://github.com/scientific-python/changelist) to
compile each pull request into an item of the release notes. Please refer to
the [instructions](https://scikit-image.org/docs/stable/development/contribute.html#documenting-changes)
and [past release notes](https://scikit-image.org/docs/stable/release_notes/index.html)
for guidance and examples.

```release-note
Clarify description of `data_range` parameter in `skimage.metrics.structural_similarity`.
```

### CODE REVIEW

**Comment by mkcor** [comment]
Uh oh, this PR should be one-commit-long, not seven... I guess that's because I mistakenly committed [216a575](https://github.com/scikit-image/scikit-image/pull/7345/commits/216a575f3aca74c63854090a81045f37bf2f0368) on (my) `main` first and this led to diverging commit histories... :woman_facepalming:

**Review by lagru — APPROVED** [review]
"difference" is a clear improvement over "distance". Feel welcome to squash and merge or tweak further if you want. :D

File: skimage/metrics/_structural_similarity.py [hunk 1]
```diff
@@ -39,8 +39,8 @@ def structural_similarity(
     gradient : bool, optional
         If True, also return the gradient with respect to im2.
     data_range : float, optional
-        The data range of the input image (distance between minimum and
-        maximum possible values). By default, this is estimated from the image
+        The data range of the input image (difference between maximum and
+        minimum possible values). By default, this is estimated from the image
         data type. This estimate may be wrong for floating-point image data.
         Therefore it is recommended to always pass this value explicitly
```
Comment (mkcor) [inline skimage/metrics/_structural_similarity.py:45]:
```suggestion
        Therefore it is recommended to always pass this scalar value explicitly
```

**Comment by mkcor** [comment]
Thank you, @lagru! I just added abbe7dfe4e68c4b8f02d6e6b75ea4159f63e4173 and I'll merge once CI is green.

**Pull request merged**